device/tt_silicon_driver.cpp

```diff
@@ -1975,14 +2033,14 @@ void tt_SiliconDevice::initialize_pcie_devices() {
     }
 
     // If requires multi-channel or doesn't support mmio-p2p, init iatus without p2p.
-    // MT Initial BH - skip device to host related init
-    if (arch_name != tt::ARCH::BLACKHOLE) {
-        if (m_num_host_mem_channels > 1 || arch_name != tt::ARCH::GRAYSKULL) {
-            init_pcie_iatus_no_p2p();
-        } else {
-            init_pcie_iatus();
-        }
+    if (m_num_host_mem_channels > 1 || arch_name != tt::ARCH::GRAYSKULL) {
```

Comment:
At first glance, it was not obvious to me which path this code takes in the case of BH and 1 memory channel.  I think there is an opportunity to improve the clarity here.

If I understand the code correctly, `init_pcie_iatus` is used only for Grayskull and a single host memory channel.  What do you think about,
```
if (m_num_host_mem_channels == 1 && arch_name == tt::ARCH::GRAYSKULL {
    init_pcie_iatus();
} else {
    ...
```

Reply:
I can't say I didn't have the same confusion at one point :) will edit this.